spring-cloud-deployer-kubernetes/src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -227,7 +227,7 @@ private boolean maxConcurrentExecutionsReached() {
 	protected String createDeploymentId(AppDeploymentRequest request) {
 		String name = request.getDefinition().getName();
 		Hashids hashids = new Hashids(name, 0, "abcdefghijklmnopqrstuvwxyz1234567890");
-		long idToEncode = System.currentTimeMillis() + ThreadLocalRandom.current().nextInt(100);
+		long idToEncode = System.nanoTime() + ThreadLocalRandom.current().nextInt(100);
```

Comment:
There seems to be not much reward for doing this. The only guarantee is that it is at least as good as System.currentTimeMillis(). Also it mentions that at some point in the future the numbers may be negative. Also, its late in the release cycle and I would prefer not to bring in another unknown.

```java
     * This method can only be used to measure elapsed time and is
     * not related to any other notion of system or wall-clock time.
     * The value returned represents nanoseconds since some fixed but
     * arbitrary <i>origin</i> time (perhaps in the future, so values
     * may be negative).  The same origin is used by all invocations of
     * this method in an instance of a Java virtual machine; other
     * virtual machine instances are likely to use a different origin.
     *
     * <p>This method provides nanosecond precision, but not necessarily
     * nanosecond resolution (that is, how frequently the value changes)
     * - no guarantees are made except that the resolution is at least as
     * good as that of {@link #currentTimeMillis()}.
```

Reply:
Then we have to increase the range of random numbers or have an AtomicLong as the number.